assignment4.Rmd

```diff
@@ -1,118 +1,168 @@
+
+COR2 <- cor(D4)
+corrplot(COR2, order="AOE", method="circle", tl.pos="lt", type="upper",        
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
+
+
+```
+## Now print out the eigenvectors (often called loadings) for the components you generated:
+
+```{r}
+pca$rotation
+
+#Examine the eigenvectors, notice that they are a little difficult to interpret. It is much easier to make sense of them if we make them proportional within each component
+
+loadings <- abs(pca$rotation) #abs() will make all eigenvectors positive
+
+F1 <- sweep(loadings, 2, colSums(loadings), "/") #sweep() computes each row as a proportion of the column. (There must be a way to do this with dplyr()?)
+
+#Now examine your components and try to come up with substantive descriptions of what some might represent?
+
+#You can generate a biplot to help you, though these can be a bit confusing. They plot the transformed data by the first two components. Therefore, the axes represent the direction of maximum variance. Then mapped onto this point cloud are the original directions of the variables, depicted as red arrows. It is supposed to provide a visualization of which variables "go together". Variables that possibly represent the same underlying construct point in the same direction.  
+
+biplot(pca)
+
+#Calculate values for each student that represent these your composite variables and then create a new correlogram showing their relationship to mean_correct.
+F2 <- cbind(F1, as.data.frame(D1$mean_correct))
+corrgram(F2)
+```
+# Part III
+
+## Also in this repository is a data set collected from class that shows how many students thought that a TC program was related to andother TC program. Use PCA to look for components of related programs. Explain why you think there are relationships between these programs.
+```{r}
+DD <- read.csv("D:/Github/4050/assignment4-master/TCprograms.csv", sep = " ")
+rownames(DD) <- DD[,1]
+DD <- DD[-69,-1]
+DD1 <- scale(DD, center = TRUE)
+pca1 <- prcomp(DD1, scale = TRUE)
+summary(pca1)
+plot(pca1, type = "lines")
+DD2 <- as.data.frame(pca1$x)
+COR3 <- cor(DD2)
+```
+
+```{r}
+pci <- summary(pca1)$importance
+pv = pci[2,]
+pca2 <- pca1$x[,-c(which(pv < 1/68))]
+pca2 <- pca2[-c(which(pv < 1/68)),]
+COR4 <- cor(pca2)
+corrplot(COR4, order="AOE", method="circle", tl.pos="lt", type="upper",        
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
+value <- pca1$x %*% t(pca1$rotation)
+```
+
+```{r}
+corrgram(pca2)
+```
+
+```{r}
+biplot(pca1, cex = 0.6)
+```
+
+
+
+#I believe there are relationships between these programs since they are accounted for more than 80% of the variance of the data. 23/68 of the data can explain 80% of the correlation, so they must highly correlated.
+

```

Comment:
You can also examine which programs are related to one another with `abs(pca$rotation)` and sorting them. But, overall, great job!